sources/web/datalab/polymer/test/item-list-test.ts

```diff
@@ -0,0 +1,211 @@
+
+  /**
+   * Returns true iff the item at the specified index is checked
+   * @param i index of item to check
+   */
+  function isSelected(i: number): boolean {
+    const row = getRow(i);
+    const box = getCheckbox(i);
+    // Check three things: the index is in the selectedIndices array, the row
+    // has the selected attribute, and its checkbox is checked
+    return testFixture.selectedIndices.indexOf(i) > -1
+        && row.hasAttribute('selected')
+        && box.checked;
+  }
+
+  /**
+   * Returns the row element at the specified index
+   * @param i index of row whose element to return
+   */
+  function getRow(i: number): HTMLElement {
+    return testFixture.$.listContainer.querySelectorAll('paper-item')[i];
+  }
+
+  /**
+   * Returns the checkbox element at the specified index
+   * @param i index of row whose checkbox element to return
+   */
+  function getCheckbox(i: number): any {
+    const row = getRow(i);
+    return row.querySelector('paper-checkbox');
+  }
+
+  /**
+   * Rows must be recreated on each test with the fixture, to avoid state
+   * leakage. Also be sure to call Polymer's flush() after any code that will
+   * cause shadow dom redistribution, such as observed array mutation, wich is
+   * used by the dom-repeater in this case.
+   */
+  beforeEach(() => {
+    testFixture = fixture('item-list-fixture');
+    const rows: ItemListRow[] = [
+      {
+        firstCol: 'first column 1',
+        icon: 'folder',
+        secondCol: 'second column 1',
+        selected: false,
+      },
+      {
+        firstCol: 'first column 2',
+        icon: 'folder',
+        secondCol: 'second column 2',
+        selected: false,
+      },
+      {
+        firstCol: 'first column 3',
+        icon: 'folder',
+        secondCol: 'second column 3',
+        selected: false,
+      }
+    ];
+    testFixture.rows = rows;
+    Polymer.dom.flush();
+  });
+
+  it('displays a row per item in list', () => {
+    const children = testFixture.$.listContainer.querySelectorAll('paper-item');
+    assert(children.length === 3, 'three rows should appear');
+  });
+
+  it('displays column names in the header row', () => {
+    testFixture.columns = ['col1', 'col2'];
+    Polymer.dom.flush();
+
+    // Column 0 is for the checkbox
+    assert(testFixture.$.header.children[1].innerText === 'col1');
+    assert(testFixture.$.header.children[2].innerText === 'col2');
+  });
+
+  it('selected items are selected', () => {
+    testFixture._selectItem(1);
+
+    assert(!isSelected(0), 'first item should not be selected');
+    assert(isSelected(1), 'second item should be selected');
+    assert(!isSelected(2), 'third item should not be selected');
+  });
+
+  it('selected items are returned', () => {
+    testFixture._selectItem(0);
+    testFixture._selectItem(2);
+
+    assert(JSON.stringify(testFixture.selectedIndices) === '[0,2]',
+        'first and third items should be selected');
+  });
+
+  it('can select/unselect all', () => {
```

Comment:
Do you want to test the behavior when some items are selected and the select-all box is clicked?